src/oge/emissions.py

```diff
@@ -378,32 +378,47 @@ def adjust_emissions_for_biomass(df):
     return df
 
 
-def calculate_co2e_mass(df, year, gwp_horizon=100, ar5_climate_carbon_feedback=True):
+def calculate_co2e_mass(df, version, gwp_horizon=100, ar5_climate_carbon_feedback=True):
     """Calculate CO2-equivalent emissions from CO2, CH4, and N2O. This is done by
     choosing one of the IPCC's emission factors for CH4 and N2O.
 
+    Note that `version` parameter controls the version of the IPCC version that will be
+    chosen. If `version` is made of numerals only then `version` is interpreted as a
+    year the most recent AR that was available in the given year is picked. If
+    `version` is mage of letters only or is alphanumeric then `version` will be
+    interpreted as the AR.
+
     If the `fuel_consumed_for_electricity_units` column is available, we also compute
     the adjusted emissions.
     """
     df_gwp = load_data.load_ipcc_gwp()
 
-    # use the most recent AR that was available in the given year
-    ipcc_version = df_gwp.loc[df_gwp["year"] <= year, "year"].max()
+    if isinstance(version, int) and version in df_gwp["year"].unique():
```

Comment:
So the year here does not need to be explicitly in the list of years - for example, if we enter version = 2022, then we'd want this to return 2021 as the most recent AR year. I think we could just remove the second test entirely and just check if it is an int.

Reply:
What about years < 1995, that will raise an error. Are we fine we that?

Reply:
In the last version of the code it raises an error if version is an `int` < the min year found in the data frame